server/src/main/java/io/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -270,4 +264,19 @@ public void cleanupCacheFiles(File baseFile, File cacheFile) throws IOException
       }
     }
   }
+
+  public Set<StorageLocation> getSortedSet(List<StorageLocation> locs)
```

Comment:
Collections.sort would require the whole list to be synchronized as we can have more than 1 downloading thread in the historicals